applications/ContactStructuralMechanicsApplication/custom_conditions/ALM_mortar_contact_condition.cpp

```diff
@@ -624,6 +625,51 @@ void AugmentedLagrangianMethodMortarContactCondition<TDim, TNumNodes, TFrictiona
 //         //                 KRATOS_WATCH(RHS_contact_pair);
 //                     LOG_VECTOR_PRETTY( RHS_contact_pair );
                 }
+                
+                // Setting the weighted gap
+                // Mortar condition matrices - DOperator and MOperator
+                const bounded_matrix<double, TNumNodes, TNumNodes>& DOperator = rThisMortarConditionMatrices.DOperator;
+                const bounded_matrix<double, TNumNodes, TNumNodes>& MOperator = rThisMortarConditionMatrices.MOperator;
+                
+                // Current coordinates 
+                const bounded_matrix<double, TNumNodes, TDim> x1 = MortarUtilities::GetCoordinates<TDim,TNumNodes>(this->GetGeometry());
+                const bounded_matrix<double, TNumNodes, TDim> x2 = MortarUtilities::GetCoordinates<TDim,TNumNodes>(mThisMasterElements[pair_index]->GetGeometry());
+        
+                const bounded_matrix<double, TNumNodes, TDim> D_x1_M_x2 = prod(DOperator, x1) - prod(MOperator, x2); 
+                
+                for (unsigned int i_node = 0; i_node < TNumNodes; i_node++)
+                {
+                    const array_1d<double, 3>& normal = GetGeometry()[i_node].GetValue(NORMAL);
+                    const array_1d<double, TDim> aux_array = row(D_x1_M_x2, i_node);
+                                    
+                    #pragma omp atomic 
```

Comment:
He element should not write to its nodes. This is very bad practice (even with the atomic)

Reply:
The weighted gap is a nodal value, any better alternative?

Reply:
I used the residual , but with the ALM is no longer possible

Reply:
Done, I used the SetLock and UnsetLock

Reply:
you did not get me right.

even using SetLock and UnsetLock (or atomic, which was correct in this case), the element/condition is assumed not to write onto its nodes. It should only read from them but not change their value.

If you need to change the nodal value than do it in the function 
AddExplicitContribution or eventually (and as a worst solution) in the function Calculate

Reply:
Add AddExplicitContribution in the base class is empty. What it does in the class where is implemented is use the SetLock and UnSetLock

Reply:
![imagen](https://user-images.githubusercontent.com/19991680/31601330-d1ff170e-b259-11e7-9720-fbd94670e740.png)
The base class:
![imagen](https://user-images.githubusercontent.com/19991680/31601368-ed380120-b259-11e7-97ca-c698737a0579.png)